roles/lib_utils/library/glusterfs_check_containerized.py

```diff
@@ -0,0 +1,187 @@
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import subprocess
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+DOCUMENTATION = '''
+---
+module: glusterfs_check_containerized
+
+short_description: Check health of each volume in glusterfs on openshift.
+
+version_added: "2.6"
+
+description:
+    - This module attempts to ensure all volumes are in healthy state
+      in a glusterfs cluster.  The module is meant to be failure-prone, retries
+      should be executed at the ansible level, they are not implemented in
+      this module.
+      This module by executing the following (roughly):
+      oc exec --namespace=<namespace> <podname> -- gluster volume list
+      for volume in <volume list>:
+        gluster volume heal <volume> info
+
+author:
+    - "Michael Gugino <mgugino@redhat.com>"
+'''
+
+EXAMPLES = '''
+- name: glusterfs volumes check
+  glusterfs_check_containerized
+    oc_bin: "/usr/bin/oc"
+    oc_conf: "/etc/origin/master/admin.kubeconfig"
+    oc_namespace: "glusterfs"
+    cluster_name: "glusterfs"
+'''
+
+
+def fail(module, err):
+    """Fail on error"""
+    result = {'failed': True,
+              'changed': False,
+              'msg': err,
+              'state': 'unknown'}
+    module.fail_json(**result)
+
+
+def call_or_fail(module, call_args):
+    """Call subprocess.check_output and return utf-8 decoded stdout or fail"""
+    try:
+        # Must decode as utf-8 for python3 compatibility
+        res = subprocess.check_output(call_args).decode('utf-8')
+    except subprocess.CalledProcessError as err:
+        fail(module, str(err))
+    return res
+
+
+def get_valid_nodes(module, oc_exec, exclude_node):
+    """Return a list of pods that might actually be running"""
+    call_args = oc_exec + ['get', 'nodes']
+    res = call_or_fail(module, call_args)
+    valid_nodes = []
+    for line in res.split('\n'):
+        fields = line.split()
+        if not fields:
+            continue
+        if fields[0] != exclude_node and fields[1] == "Ready":
+            valid_nodes.append(fields[0])
+    if not valid_nodes:
+        fail(module,
+             'Unable to find suitable node in get nodes output: {}'.format(res))
+    return valid_nodes
+
+
+def select_pod(module, oc_exec, cluster_name, valid_nodes):
+    """Select a pod to attempt to run gluster commands on"""
+    call_args = oc_exec + ['get', 'pods', '-owide']
+    res = call_or_fail(module, call_args)
+    # res is returned as a tab/space-separated list with headers.
+    res_lines = res.split('\n')
+    pod_name = None
+    name_search = 'glusterfs-{}'.format(cluster_name)
```

Comment:
@jarrpa Rather than looking for a name, is there not a label for glusterfs pods?

Reply:
IMO, name is simpler, more guaranteed to not be manipulated by users than labels.

Reply:
ok